packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -61,15 +58,12 @@ export default function markdown({ settings, logger }: AstroPluginOptions): Plug
 
 				const fileURL = pathToFileURL(fileId);
 
-				// `processor` is initialized in `buildStart`, and removed in `buildEnd`. `load`
-				// should be called in between those two lifecycles, so this error should never happen
+				// Lazily initialize the Markdown processor
 				if (!processor) {
-					return this.error(
-						'MDX processor is not initialized. This is an internal error. Please file an issue.',
-					);
+					processor = createMarkdownProcessor(settings.config.markdown);
 				}
 
-				const renderResult = await processor
+				const renderResult = await (await processor)
```

Comment:
why not await `createMarkdownProcessor` once instead?

Reply:
Since `processor` is lazily initialized, we can't await during initialization since if the transform hook runs concurrently, you could be initializing it multiple times in parallel. Making the initialization sync instead prevents this, but we'd need to await it here when using it.